src/loader/README.md

```diff
@@ -16,6 +16,44 @@ version 0.8.0 will have timescaledb-0.8.0.so loaded.
    schedulers for any database that has TimescaleDB installed.
    This is done by a background task called the launcher.
 
+# Messages the launcher may receive
+The launcher implements a simple message queue to be notified when
+it should take certain actions, like starting or restarting a
+scheduler for a given database. Note that on startup of the server
+or on creation of a database, the launcher will attempt to start 
+a scheduler for each database, as it cannot figure out whether a 
+scheduler should exist for a given database. The scheduler is 
+responsible for shutting down if it should not exist.
+
+##Message types, when and why they are sent to the launcher:
+
+`start`: Used to start the scheduler by the user, it is meant to be
+an idempotent start, as in, if it is run multiple times, and startup
+is successful, it is the same as if it were run once. It is used
+mainly to reactivate a scheduler that the user had stopped.
+
+`stop`: Used to stop the scheduler in the case of a drop database 
+command, databases cannot be dropped if there are any open connections.
+The `stop` message is idempotent. 
+
+`restart`: Used to stop and restart the scheduler if it is running
+either when the extension is updated (when updating from one version
+with a scheduler to another) or dropped. The scheduler is immediately
+restarted, but waits on the vxid of the txn that sent the message, 
+therefore it will shut down if the extension is dropped and will 
+run the proper version if there is a rollback or commit on either of
+the other commands. It is not idempotent. 
+
+`start_or_restart`: Used to either stop and restart the scheduler or 
+start it if it does not exist. This is used when the extension is 
+created or when it is updated from a version without a scheduler 
+to a version with a scheduler (0.12 and up). The start or restart
+action avoids a possible race condition where a scheduler had been 
+started, but not given a vxid (or was waiting on an incorrect vxid
+so a `start` message might not see the correct version of TimescaleDB.
+A restart will not work, because in many cases the scheduler will not
```

Comment:
The sentence beginning `A restart will not work...` belongs in a separate section about the specific 0.11.0-->0.12.0 upgrade.